### PR TITLE
UGENE-7052. Copied MSA region should be pasted with no sequence name

### PR DESCRIPTION
**Pull request by yalgaer**

### CODE REVIEW

File: src/corelibs/U2Core/src/globals/ClipboardController.cpp [hunk 1]
```diff
@@ -26,6 +26,8 @@
 
 namespace U2 {
 
+const QString U2Clipboard::ugeneMimeType = "text/ugene";
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Core/src/globals/ClipboardController.cpp]: `static const QString U2Clipboard::UGENE_MIME_TYPE = "text/ugene";`

**Review by DmitriiSukhomlinov — DISMISSED** [review]

File: src/corelibs/U2Core/src/globals/ClipboardController.cpp (hunk 1 above)
Comment (yalgaer) [reply src/corelibs/U2Core/src/globals/ClipboardController.cpp]: Fixed.

**Review by evelinabis — DISMISSED** [review]

File: src/corelibs/U2Gui/src/PasteController.cpp [hunk 2]
```diff
@@ -106,18 +106,19 @@ PasteTask *PasteFactoryImpl::createPasteTask(bool isAddToProject) {
...
         return new PasteUrlsTask(mimeData->urls(), isAddToProject);
     }
-    QString clipboardText;
     try {
-        clipboardText = clipboard->text();
+        QString clipboardText = mimeData->hasFormat(U2Clipboard::UGENE_MIME_TYPE)
+                                    ? QString::fromUtf8(mimeData->data(U2Clipboard::UGENE_MIME_TYPE))
+                                    : clipboard->text();
+        if (clipboardText.isEmpty()) {
+            coreLog.error("UGENE can not recognize current clipboard content as one of the supported formats.");
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Gui/src/PasteController.cpp]: tr

File: src/corelibs/U2Gui/src/PasteController.cpp [hunk 3]
```diff
@@ -106,18 +106,19 @@ PasteTask *PasteFactoryImpl::createPasteTask(bool isAddToProject) {
...
+            return nullptr;
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Gui/src/PasteController.cpp:115]: I don't like the structure here. The natural way out from `try` is `throw`, not `return`. Also I don't see any objects which are able to throw the exception at all, so the reason of the `try` block here is dubious. I offer you whether throw an exception here and catch it in the corresponding block or get rid of "try-catch" at all.

File: src/corelibs/U2Gui/src/PasteController.cpp [hunk 4]
```diff
@@ -106,18 +106,19 @@ PasteTask *PasteFactoryImpl::createPasteTask(bool isAddToProject) {
...
+        }
+        return new PasteTextTask(clipboardText, isAddToProject);
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Gui/src/PasteController.cpp:117]: OK, not at all, `new` are able to throw `bad_alloc`, but I think it's OK if we don't take it an account.

**Review by DmitriiSukhomlinov — CHANGES_REQUESTED** [review]
Some comments are need to be resolved

File: src/corelibs/U2Gui/src/PasteController.cpp (hunk 2 above)
Comment (yalgaer) [reply src/corelibs/U2Gui/src/PasteController.cpp]: Fixed

File: src/corelibs/U2Gui/src/PasteController.cpp (hunk 3 above)
Comment (yalgaer) [reply src/corelibs/U2Gui/src/PasteController.cpp:115]: This behavior: ("Data in clipboard is too large") was not changed in this PR at all.
We can create another issue to investigate why it was added here originally as a part of separate bug.

Comment (DmitriiSukhomlinov) [reply src/corelibs/U2Gui/src/PasteController.cpp:115]: OK, [done](https://ugene.dev/tracker/browse/UGENE-7060).

**Review by DmitriiSukhomlinov — APPROVED** [review]

**Pull request merged**